staNMF/staNMF.py

```diff
@@ -371,10 +392,15 @@ def instability(self, k1=0, k2=0):
                 self.instabilitydict[k] = (np.sum(distMat) / (numReplicates *
                                            (numReplicates-1)))
 
-                outputfile = open("instability.csv", "w")
-                for i in sorted(self.instabilitydict):
-                    outputfile.write(str(i) + "," +
-                                     str(self.instabilitydict[i]) + "\n")
+                if self.parallel:
+                    outputfile = open(str(path + "instability.csv"), "w")
+                    outputfile.write("\n" + str(k) + "," + str(
```

Comment:
Does `outputfile` ever get closed?

It's probably not going to cause a bug to not close the file, but it's best practice, and may matter in some cases (like if you're using compression I think).